Library/Homebrew/cmd/autoremove.rb

```diff
@@ -0,0 +1,68 @@
+# frozen_string_literal: true
+
+require "formula"
+require "cli/parser"
+
+module Homebrew
+  module_function
+
+  def autoremove_args
+    Homebrew::CLI::Parser.new do
+      usage_banner <<~EOS
+        `autoremove` [<options>]
+
+        Remove packages that weren't installed on request and are no longer needed.
+      EOS
+      switch "-n", "--dry-run",
+             description: "Just print what would be removed."
+      switch "-f", "--force",
+             description: "Remove without confirmation."
+      named 0
+    end
+  end
+
+  def get_removable_formulae(installed_formulae)
+    removable_formulae = []
+
+    installed_formulae.each do |formula|
+      # Reject formulae installed on request.
+      next if formula.installed_kegs.any? { |keg| Tab.for_keg(keg).installed_on_request }
```

Comment:
Does this need to go through all installed kegs or can it just check the most recent one?

Reply:
Good question, I'm not sure about that.

Reply:
I'd say it can just look at the most recent one for performance. That means `formula.tab.installed_on_request ` should work as you expect iirc.